src/components/App.js

```diff
@@ -1,15 +1,161 @@
-    <img
-      src="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg"
-      alt="react-logo"
-    />
-    <Weather />
-  </div>
-);
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      location: false,
+      latitude: null,
+      longitude: null,
+      weatherData: {},
+      forecastData: {},
+      forecastResponse: false,
+      weatherResponse: false,
+      uvResponse: false,
+      uvData: null,
+      temperature: {},
+    };
+
+    this.changeUnit = this.changeUnit.bind(this);
+  }
+
+  componentDidMount() {
+    if ('geolocation' in navigator) {
+      getLocation().then(response => {
+        this.setState({
+          location: true,
+          latitude: response.coords.latitude,
+          longitude: response.coords.longitude,
+        });
+        this.getCurrentWeather();
+        this.getCurrentForecast();
+        this.getUVIndex();
+      });
+    }
+  }
+
+  getCurrentWeather() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('weather', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          weatherResponse: true,
+          temperature: { type: 'FAHRENHEIT', value: data.main.temp, unit: 'F' },
+          weatherData: {
+            name: data.name,
+            description: data.weather[0].description,
+            icon: data.weather[0].icon,
+          },
+        });
+      });
+    }
+  }
+
+  getCurrentForecast() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('forecast/daily', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          forecastResponse: true,
+          forecastData: data.list,
+        });
+      });
+    }
+  }
+
+  getUVIndex() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('uvi', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          uvData: data,
+          uvResponse: true,
+        });
+      });
+    }
+  }
+
+  changeUnit(unit) {
+    const { temperature } = this.state;
+    if (unit === 'FAHRENHEIT') {
+      const res = toFahrenheit(temperature);
+      this.setState({
+        temperature: res,
+      });
+    } else {
+      const res = toCelcius(temperature);
+      this.setState({
+        temperature: res,
+      });
+    }
+  }
+
+  render() {
+    const {
+      weatherData,
+      forecastData,
+      forecastResponse,
+      weatherResponse,
+      uvResponse,
+      uvData,
+      temperature,
+    } = this.state;
+    return (
+      <div className="container">
+        <BrowserRouter>
+          <div className="weather-container">
+            <div className="weather-container__content">
+              <UnitToggle unitChange={this.changeUnit} />
```

Comment:
keep your prop/state/function names consistent. you'll thank yourself later when you go looking for the origin of the function.

Reply:
done